### PR TITLE
[CALCITE-4295] Composite of two checkers with SqlOperandCountRange throws IllegalArgumentException

### PR DESCRIPTION
**Pull request by docete**
Root cause is:
The minMin function of CompositeOperandTypeChecker should return the minimum of lower bound of all SqlOperandCountRange.This PR will fix it.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java [hunk 1]
```diff
@@ -4048,6 +4048,18 @@ private Tester getExtendedTester() {
     sql(sql).trim(true).ok();
   }
 
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4295">[CALCITE-4295]
+   * Composite of two checker with SqlOperandCountRange throws IllegalArgumentException</a>.
```
Comment (rubenada) [inline core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java:4054]: The title in the comment reflects the "old" title of the ticket, could it be updated?

**Comment by rubenada** [comment]
@docete Could you please fix the grammar in the commit message (and title in Jira), e.g. something like: "The minMin function of CompositeOperandTypeChecker should return the minimum lower bound of all SqlOperandCountRange"
Also, please make sure that you squash all commits into a single one, and you add your name between parenthesis at the end of the commit message. I think we could squeeze this into next version 1.26 coming soon. Thanks

**Comment by rubenada** [comment]
Thanks @docete . The patch LGTM
@danny0405 I saw you comment in the Jira ticket, could you please take a look at this PR?

**Comment by docete** [comment]
@rubenada I updated the JIRA ticket and commit message.

**Pull request merged**